python/Ganga/Core/GangaRepository/Registry.py

```diff
@@ -691,7 +691,8 @@ def _flush(self, objs):
 
         for obj in objs:
             # check if the object is dirty, if not do nothing
-            if not obj._dirty:
+            this_id = self.find(obj)
+            if not obj._dirty or this_id not in self._loaded_ids:
```

Comment:
Indeed - I'd be very worried if we can have jobs that can be marked dirty but **not** loaded! Do we have an example of this somewhere?